### PR TITLE
MAINT: Add more pytest idiom to parser tests

### PR DESCRIPTION
**Pull request by gfyoung**
This is only a start.  To keep the diff small, we are only converting `c_parser_only`.

If this looks good, the other test classes will soon follow in subsequent PR's.

cc @jreback

### CODE REVIEW

**Comment by jreback** [comment]
how do you run all of the tests on the c parser though? don't you need to also inject the fixture into the current test_parser.py?

File: pandas/tests/io/parser/conftest.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import pytest
+
+from pandas import read_csv, read_table
+import pandas.util.testing as tm
+
+
+class BaseParser(object):
```
Comment (jbrockmendel) [inline pandas/tests/io/parser/conftest.py]: wouldn't these usually go in a base.py or something?  I guess if you need them for the fixtures you could import them from there?

Comment (gfyoung) [reply pandas/tests/io/parser/conftest.py]: `base.py` semantically has generally been used for importing into other test classes.  I'm not sure I see the need to separate out all of configuration so to speak.  The fixtures in `conftest.py` will expose them appropriately once all test classes have been converted.

**Comment by gfyoung** [comment]
@jreback @jbrockmendel : Here is the plan:

* `c_parser_only.py` is now separate from `test_parsers.py`, so you can just run `pyest test_c_parser_only.py` (the fixtures get automatically injected per the newly created `conftest.py`).  The new format of those tests is 100% functional (no classes), which is more `pytest`-idiomatic.

* If this format looks good, I will convert all other files (e.g. `header.py`, `comment.py`, `mangle_dupe_cols.py` similarly in subsequent PR's.  Once that is done, we no longer need `test_parsers.py`.  All tests will be nice and modular and not all clumped into one giant test class as they are currently.

Hopefully that clears things up?

**Comment by gfyoung** [comment]
@jreback : Any further thoughts on this?

**Comment by jreback** [comment]
lgtm @gfyoung keep em coming!

**Pull request merged**